openstack/clientconfig/requests.go

```diff
@@ -749,6 +753,44 @@ func NewServiceClient(service string, opts *ClientOpts) (*gophercloud.ServiceCli
 		return nil, err
 	}
 
+	// Check if a custom CA cert was provided.
+	// First, check if the CACERT environment variable is set.
+	var caPath string
+	var caCert string
+	if v := env.Getenv(envPrefix + "CACERT"); v != "" {
+		caPath = v
+	}
+
+	// Next, check if the cloud entry sets a CA cert.
+	if v := cloud.CACertFile; v != "" {
+		caPath = v
+	}
+
+	// Get the ca-cert-bundle key if there is a value for CA cert.
+	if caPath != "" {
+		caFile, err := ioutil.ReadFile(caPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read CA cert from disk: %s", err)
+		}
+		caCert = string(bytes.TrimSpace(caFile))
```

Comment:
You're converting bytes to a string, then convert a string to bytes on the line 784. Can you define the `caCert` var as `[]byte` initially?